speechbrain/utils/bertscore.py

```diff
@@ -0,0 +1,371 @@
+
+    Special tokens (as queried from the tokenizer) are entirely ignored.
+
+    Authors' reference implementation of the metric can be found
+    `here <https://github.com/Tiiiger/bert_score>`_. The linked page extensively
+    describes the approach and compares how the BERTScore relates to human
+    evaluation with many different models.
+
+    .. warning::
+        Out of the box, this implementation may not strictly match the results
+        of the reference implementation. Please read the argument documentation
+        to understand the differences.
+
+    Arguments
+    ---------
+    model : transformers.AutoModelForTextEncoding
+        Transformers text encoder. May live on a non-CPU device.
+    tokenizer : transformers.AutoTokenizer
+        Transformers tokenizer.
+    batch_size : int, optional
+        How many pairs of utterances should be considered at once. Higher is
+        faster but may result in OOM.
+    use_idf : bool, optional
+        If enabled (default), tokens in the reference are weighted by
+        Inverse Document Frequency, which allows to weight down the impact of
+        common words that may carry less information. Every sentence appended
+        is considered a document in the IDF calculation.
+    sentence_level_averaging : bool, optional
+        When `True`, the final recall/precision metrics will be the average of
+        recall/precision for each tested sentence, rather of each tested token,
+        e.g. a very long sentence will weigh as much as a very short sentence in
+        the final metrics. The default is `True`, which matches the reference
+        implementation.
+    num_layers : int, optional
+        When specified, and assuming the passed LM can be truncated that way,
+        the encoder for the passed model will be truncated to the specified
+        layer (mutating it). This means that the embeddings will be those of the
+        Nth layer rather than the last layer. The last layer is not necessarily
+        the best when it comes to the quality of the metric. Note that the
+        reference implementation automatically selects a default hardcoded layer
+        for certain models.
+    allow_matching_special_tokens : bool, optional
+        When `True`, non-special tokens may match against special tokens during
+        greedy matching (e.g. `[CLS]`/`[SEP]`). Batch size must be 1 due to
+        padding handling.
+        The default is `False`, which is different behavior from the reference
+        implementation (see
+        `bert_score#180 <https://github.com/Tiiiger/bert_score/issues/180>`_).
+    """
+
+    def __init__(
+        self,
+        lm,
+        tokenizer,
+        batch_size: int = 64,
+        use_idf: bool = True,
+        sentence_level_averaging: bool = True,
+        num_layers: Optional[int] = None,
+        allow_matching_special_tokens: bool = False,
+    ):
+        self.clear()
+        self.lm = lm
+        self.tokenizer = tokenizer
```

Comment:
okay what about using our HFTransformer interface for loading the LM/tokenizer automatically?